### PR TITLE
Fix calculation of Windows kernel base address for NT build 16299

### PR DESCRIPTION
**Pull request by fwruck**
Fix calculation of Windows kernel base address for Windows 10.0.16299.15 /
    Add a third way to calculate the windows kernel base address
    (Get get base address using lstar/cstar and KiSystemCall64Shadow/KiSystemCall32Shadow)
    
     On branch fixwinkidivideerrorfault
     Changes to be committed:
            modified:   libvmi/os/windows/core.c

Based on the work of @kirschju

### CODE REVIEW

File: libvmi/os/windows/core.c [hunk 1]
```diff
@@ -698,7 +701,30 @@ init_from_rekall_profile(vmi_instance_t vmi)
             windows->ntoskrnl_va = kpcr - kpcr_rva;
             if ( VMI_FAILURE == vmi_translate_kv2p(vmi, windows->ntoskrnl_va, &windows->ntoskrnl) )
                 goto done;
+		//Get kernel base address using cstar/lstar and KiSystemCall32Shadow / KiSystemCall64Shadow 
+        } else if ( VMI_SUCCESS == rekall_profile_symbol_to_rva(windows->rekall_profile, "KiSystemCall64Shadow", NULL, &int0_rva) ) {
```
Comment (tklengyel) [inline libvmi/os/windows/core.c]: If you already grab `KiSystemCall64Shadow` here..

File: libvmi/os/windows/core.c [hunk 2]
```diff
@@ -698,7 +701,30 @@ init_from_rekall_profile(vmi_instance_t vmi)
...
 
+            vmi_get_vcpureg(vmi, &lstar, MSR_LSTAR, 0);
+            vmi_get_vcpureg(vmi, &cstar, MSR_CSTAR, 0);
+
+
+    		rekall_profile_symbol_to_rva(windows->rekall_profile, "KiSystemCall32Shadow", NULL, &kisystemcall32shadow);
+		    rekall_profile_symbol_to_rva(windows->rekall_profile, "KiSystemCall64Shadow", NULL, &kisystemcall64shadow);
```
Comment (tklengyel) [inline libvmi/os/windows/core.c]: .. there is no point grabbing it here again.

File: libvmi/os/windows/core.c [hunk 3]
```diff
@@ -698,7 +701,30 @@ init_from_rekall_profile(vmi_instance_t vmi)
...
             if ( VMI_FAILURE == vmi_translate_kv2p(vmi, windows->ntoskrnl_va, &windows->ntoskrnl) )
                 goto done;
+		//Get kernel base address using cstar/lstar and KiSystemCall32Shadow / KiSystemCall64Shadow 
+        } else if ( VMI_SUCCESS == rekall_profile_symbol_to_rva(windows->rekall_profile, "KiSystemCall64Shadow", NULL, &int0_rva) ) {
 
+            vmi_get_vcpureg(vmi, &lstar, MSR_LSTAR, 0);
+            vmi_get_vcpureg(vmi, &cstar, MSR_CSTAR, 0);
+
+
+    		rekall_profile_symbol_to_rva(windows->rekall_profile, "KiSystemCall32Shadow", NULL, &kisystemcall32shadow);
```
Comment (tklengyel) [inline libvmi/os/windows/core.c]: Should check the return of this call

**Comment by tklengyel** [comment]
@libvmi-jenkins This is ok to test.

File: libvmi/os/windows/core.c (hunk 2 above)
Comment (fwruck) [reply libvmi/os/windows/core.c]: Youre absolutely right, i'm changing this

**Comment by fwruck** [comment]
Thank you for your comments, i commited the changes

**Comment by tklengyel** [comment]
There are more style-issues as shown at: https://travis-ci.org/libvmi/libvmi/jobs/332930474

**Comment by fwruck** [comment]
Sorry for that, now all style issues are fixed

**Comment by smaresca** [comment]
@fwruck please fixup the commit history down to one commit

**Comment by fwruck** [comment]
I squashed all commits into one

**Comment by tklengyel** [comment]
Thanks, this looks good to me!

**Comment by fwruck** [comment]
Your welcome,
thank you for providing libvmi !

**Pull request merged**